File: frontend/src/types/documents.ts
export type DocumentStatus =
  | "idle"
  | "uploading"
  | "processing"
  | "ready"
  | "failed"
  
export type DocumentItem = {
  id: string
  title: string
  filename: string
  status: string
  created_at?: string
  updated_at?: string
  text_content?: string | null

  owner_user_id?: string | null
  path?: string
  file_type?: string
  scope?: string
  source_type?: string
  kb?: string
  mime_type?: string | null
  file_size?: number
  last_error?: string | null
  file_hash?: string | null

  chunk_count?: number
  document_chunks?: Array<{
    id: string
    content?: string
    status?: string
    order?: number
    section_title?: string
    doc_title?: string
  }>

  chunking_config?: {
    mode?: "auto" | "markdown" | "fixed" | string
    requested_mode?: "auto" | "markdown" | "fixed" | string
    effective_mode?: "markdown" | "fixed" | string | null
    chunk_size?: number
    overlap?: number
  }
}

File: backend/app/schemas/documents.py
from typing import Optional, List
from pydantic import BaseModel


class ChunkingConfigRead(BaseModel):
    requested_mode: Optional[str] = None
    effective_mode: Optional[str] = None
    chunk_size: Optional[int] = None
    overlap: Optional[int] = None


class DocumentRead(BaseModel):
    id: str
    title: str
    filename: str
    status: str

    owner_user_id: Optional[str] = None
    path: Optional[str] = None
    file_type: Optional[str] = None
    scope: Optional[str] = None
    source_type: Optional[str] = None
    kb: Optional[str] = None

    mime_type: Optional[str] = None
    file_size: Optional[int] = None
    file_hash: Optional[str] = None

    text_content: Optional[str] = None
    created_at: Optional[str] = None
    updated_at: Optional[str] = None
    last_error: Optional[str] = None

    chunking_config: Optional[ChunkingConfigRead] = None


class DocumentUploadResponse(BaseModel):
    document: DocumentRead